api/src/integration/java/org/pmiops/workbench/ApplicationTest.java

```diff
@@ -0,0 +1,239 @@
+import org.pmiops.workbench.api.ConfigController;
+import org.pmiops.workbench.api.DomainLookupService;
+import org.pmiops.workbench.api.ProfileController;
+import org.pmiops.workbench.api.StatusController;
+import org.pmiops.workbench.api.WorkspacesController;
+import org.pmiops.workbench.auth.ProfileService;
+import org.pmiops.workbench.auth.UserInfoService;
+import org.pmiops.workbench.blockscore.BlockscoreService;
+import org.pmiops.workbench.cdr.CdrDbConfig;
+import org.pmiops.workbench.cdr.dao.CriteriaDao;
+import org.pmiops.workbench.cohortbuilder.ParticipantCounter;
+import org.pmiops.workbench.cohortbuilder.QueryBuilderFactory;
+import org.pmiops.workbench.cohortbuilder.querybuilder.CodesQueryBuilder;
+import org.pmiops.workbench.cohortbuilder.querybuilder.DemoQueryBuilder;
+import org.pmiops.workbench.cohortreview.CohortReviewService;
+import org.pmiops.workbench.cohorts.CohortMaterializationService;
+import org.pmiops.workbench.config.BigQueryConfig;
+import org.pmiops.workbench.config.CacheSpringConfiguration;
+import org.pmiops.workbench.config.CdrConfig;
+import org.pmiops.workbench.config.CommonConfig;
+import org.pmiops.workbench.config.ConceptCacheConfiguration;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.config.WorkbenchEnvironment;
+import org.pmiops.workbench.db.WorkbenchDbConfig;
+import org.pmiops.workbench.db.dao.CdrVersionDao;
+import org.pmiops.workbench.db.dao.CohortAnnotationDefinitionDao;
+import org.pmiops.workbench.db.dao.CohortDao;
+import org.pmiops.workbench.db.dao.CohortReviewDao;
+import org.pmiops.workbench.db.dao.CohortService;
+import org.pmiops.workbench.db.dao.ConfigDao;
+import org.pmiops.workbench.db.dao.ParticipantCohortAnnotationDao;
+import org.pmiops.workbench.db.dao.ParticipantCohortStatusDao;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.dao.UserService;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.dao.WorkspaceService;
+import org.pmiops.workbench.firecloud.FireCloudConfig;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.google.CloudStorageService;
+import org.pmiops.workbench.google.DirectoryService;
+import org.pmiops.workbench.interceptors.AuthInterceptor;
+import org.pmiops.workbench.interceptors.ClearCdrVersionContextInterceptor;
+import org.pmiops.workbench.interceptors.CronInterceptor;
+import org.pmiops.workbench.mailchimp.MailChimpService;
+import org.pmiops.workbench.notebooks.NotebooksConfig;
+import org.pmiops.workbench.notebooks.NotebooksService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {TestWebMvcConfig.class})
+public class ApplicationTest {
+
+    @Autowired
```

Comment:
Do we have to list all these out here? Is it possible for us to programatically detect all bindings for services / controllers / etc. and test them? (We are likely to forget to add new stuff here.)

Reply:
👍 It would be nice to not have to update this for all new controllers/services/etc.

Reply:
So it seems that Spring doesn't have a handy way to get beans with certain annotations like @Service or @Configuration. There is a way to get all bean definitions and iterate them. I'm working on a solution for this. It won't completely erase the need to manage this test but it will definitely minimized the affects when beans are added.

Reply:
Test now loads: @NoRepositoryBean, @Service, @Controller, @Component and @Configuration classes